OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -2361,22 +2361,65 @@ public static DreamValue NativeProc_oviewers(NativeProc.Bundle bundle, DreamObje
         [DreamProc("splittext")]
         [DreamProcParameter("Text", Type = DreamValueTypeFlag.String)]
         [DreamProcParameter("Delimiter", Type = DreamValueTypeFlag.String)]
+        [DreamProcParameter("Start", Type = DreamValueTypeFlag.Float, DefaultValue = 1)]
+        [DreamProcParameter("End", Type = DreamValueTypeFlag.Float, DefaultValue = 0)]
+        [DreamProcParameter("include_delimiters", Type = DreamValueTypeFlag.Float, DefaultValue = 0)]
         public static DreamValue NativeProc_splittext(NativeProc.Bundle bundle, DreamObject? src, DreamObject? usr) {
             if (!bundle.GetArgument(0, "Text").TryGetValueAsString(out var text)) {
                 return new DreamValue(bundle.ObjectTree.CreateList());
             }
 
-            var arg2 = bundle.GetArgument(1, "Delimiter");
-            if (!arg2.TryGetValueAsString(out var delimiter)) {
-                if (!arg2.Equals(DreamValue.Null)) {
-                    return new DreamValue(bundle.ObjectTree.CreateList());
+            int start = 0;
+            int end = 0;
+            if(bundle.GetArgument(2, "Start").TryGetValueAsInteger(out start))
+                start -= 1; //1-indexed
+            if(bundle.GetArgument(3, "End").TryGetValueAsInteger(out end))
+                if(end == 0)
+                    end = text.Length;
+                else
+                    end -= 1; //1-indexed
+            bool includeDelimiters = false;
+            if(bundle.GetArgument(4, "include_delimiters").TryGetValueAsInteger(out var include_delimiters_int))
+                includeDelimiters = include_delimiters_int != 0; //idk why BYOND doesn't just use truthiness, but it doesn't, so...
```

Comment:
Rename to `includeDelimitersInt`
```suggestion
            if(bundle.GetArgument(4, "include_delimiters").TryGetValueAsInteger(out var includeDelimitersInt))
                includeDelimiters = includeDelimitersInt != 0; //idk why BYOND doesn't just use truthiness, but it doesn't, so...
```